src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -184,13 +193,16 @@ def list_role_assignments(cmd, assignee=None, role=None, resource_group_name=Non
     # fill in principal names
     principal_ids = set(i['properties']['principalId'] for i in results if i['properties']['principalId'])
     if principal_ids:
-        principals = _get_object_stubs(graph_client, principal_ids)
-        principal_dics = {i.object_id: _get_displayable_name(i) for i in principals}
+        try:
+            principals = _get_object_stubs(graph_client, principal_ids)
+            principal_dics = {i.object_id: _get_displayable_name(i) for i in principals}
 
-        for i in [r for r in results if not r['properties'].get('principalName')]:
-            i['properties']['principalName'] = ''
-            if principal_dics.get(i['properties']['principalId']):
-                i['properties']['principalName'] = principal_dics[i['properties']['principalId']]
+            for i in [r for r in results if not r['properties'].get('principalName')]:
+                i['properties']['principalName'] = ''
+                if principal_dics.get(i['properties']['principalId']):
+                    i['properties']['principalName'] = principal_dics[i['properties']['principalId']]
+        except (CloudError, GraphErrorException):
+            pass   # failure on resolving principal due to graph permission should not block the whole thing
```

Comment:
worth log a warning?

Reply:
Let me log an info. It is an expected exception when login as a SP w/p graph permission, so warning might be too much.